ffi/src/lib.rs

```diff
@@ -37,6 +37,7 @@ pub use domain_metadata::get_domain_metadata;
 pub mod engine_data;
 pub mod engine_funcs;
 pub mod error;
+pub mod table_changes;
```

Comment:
Since ArrowFFIData is gated behind a flag, I think we need to gate this too:
```suggestion
#[cfg(feature = "default-engine-base")]
pub mod table_changes
```